Content.Server/Administration/UI/BanUsernamePanelEui.cs

```diff
@@ -0,0 +1,183 @@
+using Content.Server.Administration.Managers;
+using Content.Server.EUI;
+using Content.Shared.Eui;
+using Content.Shared.Administration;
+
+using UsernameHelpers = Robust.Shared.AuthLib.UsernameHelpers;
+
+namespace Content.Server.Administration;
+
+public sealed class BanUsernamePanelEui : BaseEui
+{
+    [Dependency] private readonly IUsernameRuleManager _usernameRuleManager = default!;
+    [Dependency] private readonly ILogManager _log = default!;
+    [Dependency] private readonly IAdminManager _admins = default!;
+
+    private readonly ISawmill _sawmill;
+
+    public BanUsernamePanelEui()
+    {
+        IoCManager.InjectDependencies(this);
+        _sawmill = _log.GetSawmill("admin.username_bans_eui");
+        _usernameRuleManager.UpdatedCache += SendSingleRule;
+    }
+
+    public override EuiStateBase GetNewState()
+    {
+        var hasBan = _admins.HasAdminFlag(Player, AdminFlags.Ban);
+        var hasHost = _admins.HasAdminFlag(Player, AdminFlags.Host);
+        return new BanUsernamePanelEuiState(hasBan, hasHost);
```

Comment:
The client already knows its admin flags, you don't have to send these manually.